tests/test_getPlotDatasetTypeNames.py

```diff
@@ -0,0 +1,80 @@
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+from __future__ import annotations
+
+import os
+from unittest import TestCase, main
+
+import lsst.utils.tests
+from lsst.analysis.tools.analysisPlots.skyObject import SkyObjectHistPlot
+from lsst.analysis.tools.tasks.objectTableTractAnalysis import ObjectTableTractAnalysisConfig
+from lsst.analysis.tools.tasks.reconstructor import getPlotDatasetTypeNames
+from lsst.daf.butler import Butler, DatasetType
+from lsst.daf.butler.tests.utils import makeTestTempDir, removeTestTempDir
+from lsst.pex.config import Config
+
+TESTDIR = os.path.abspath(os.path.dirname(__file__))
+
+
+class TestGetPlotDatasetTypeNames(TestCase):
+    """Test plot dataset type names are correctly queried by the butler."""
+
+    def setUp(self):
+        self.root = makeTestTempDir(TESTDIR)
+        Butler.makeRepo(self.root)
+        self.butler = Butler(self.root, run="test_run")
+        # No dimensions in dataset type so we don't have to worry about
+        # inserting dimension data or defining data IDs.
+        self.datasetTypePlot = DatasetType(
+            "plotType_config", dimensions=(), storageClass="Config", universe=self.butler.registry.dimensions
+        )
+        self.butler.registry.registerDatasetType(self.datasetTypePlot)
+        self.datasetTypeOther = DatasetType(
+            "otherType_config", dimensions=(), storageClass="Config", universe=self.butler.registry.dimensions
+        )
+        self.butler.registry.registerDatasetType(self.datasetTypeOther)
+
+    def tearDown(self):
+        removeTestTempDir(self.root)
+
+    def testGetPlotNames(self) -> None:
+        """Test that we can get plot dataset type names from a butler."""
+        plotConfig = ObjectTableTractAnalysisConfig()
+        plotConfig.plots.skyObjectHistPlot = SkyObjectHistPlot()
+        otherConfig = Config()
+        _ = self.butler.put(plotConfig, "plotType_config")
+        _ = self.butler.put(otherConfig, "otherType_config")
+        plots = getPlotDatasetTypeNames(self.butler, "test_run")
+        for plot in plots:
+            self.assertIn("skyObjectHistPlot", plot)
```

Comment:
I'm confused by this loop: `getPlotDatasetTypeNames` should return a list with exactly one item, so you should be able to do `self.assertEqual(["skyObjectHistPlot", plots], with no loop, no?

Reply:
Try running it yourself and you will see, by default, that this list contains 6 elements: one for each band.

Reply:
Oh, hmm... In that case I'd probably make that the explicit list I'm testing against with `assertEquals`. You know exactly what the output should be, just test it directly.

Reply:
What benefit does this have relative to its current set up? I can't see it myself, and as this unit test is already relatively complex, I'm inclined to leave this as-is.

Reply:
When you know the *exact* output that a function should have, you should test that exact output. Currently, there are a variety of things that would pass this test, but aren't actually what you're expecting (for example, `["g_skyObjectHistPlot", "g_skyObjectHistPlot"]` would pass, but I'm pretty sure is incorrect). It actually simplifies the test code, too: there's no loop with `assertIn` just an explicit "here's what the answer should be".

Reply:
In this case @leeskelvin is right to leave it as an `assertIn`, as the exact bands depend on what the default configs of the task are, which is outside the unit test. One could go though and rigorously construct everything such that all inputs are known exactly, but it adds nothing to the concreteness of the test for the behavior in the function, and increases the complexity, and future maintenance of the test.